pkg/controller/docdb/dbsubnetgroup/setup.go

```diff
@@ -0,0 +1,203 @@
+	e.preCreate = preCreate
+	e.postCreate = postCreate
+	e.preDelete = preDelete
+	e.filterList = filterList
+}
+
+type hooks struct {
+	client docdbiface.DocDBAPI
+	kube   client.Client
+}
+
+func preObserve(_ context.Context, cr *svcapitypes.DBSubnetGroup, obj *svcsdk.DescribeDBSubnetGroupsInput) error {
+	obj.DBSubnetGroupName = awsclient.String(meta.GetExternalName(cr))
+	return nil
+}
+
+func postObserve(_ context.Context, cr *svcapitypes.DBSubnetGroup, resp *svcsdk.DescribeDBSubnetGroupsOutput, obs managed.ExternalObservation, err error) (managed.ExternalObservation, error) {
+	if err != nil {
+		return obs, err
+	}
+
+	cr.SetConditions(v1.Available())
+	return obs, nil
+}
+
+func (e *hooks) isUpToDate(cr *svcapitypes.DBSubnetGroup, resp *svcsdk.DescribeDBSubnetGroupsOutput) (bool, error) {
+	group := resp.DBSubnetGroups[0]
+
+	if meta.GetExternalName(cr) != awsclient.StringValue(cr.Spec.ForProvider.DBSubnetGroupName) {
+		return false, errors.New(errModifyDBSubnetGroupName)
+	}
+
+	switch {
+	case awsclient.StringValue(cr.Spec.ForProvider.DBSubnetGroupDescription) != awsclient.StringValue(group.DBSubnetGroupDescription),
+		!areSubnetsEqual(cr.Spec.ForProvider.SubnetIDs, group.Subnets):
+		return false, nil
+	}
+
+	return svcutils.AreTagsUpToDate(e.client, cr.Spec.ForProvider.Tags, group.DBSubnetGroupArn)
+}
+
+func areSubnetsEqual(specSubnetIds []*string, current []*svcsdk.Subnet) bool {
+	if len(specSubnetIds) != len(current) {
+		return false
+	}
+
+	currentMap := make(map[string]*svcsdk.Subnet, len(current))
+	for _, s := range current {
+		currentMap[awsclient.StringValue(s.SubnetIdentifier)] = s
+	}
+
+	for _, spec := range specSubnetIds {
+		if _, exists := currentMap[awsclient.StringValue(spec)]; !exists {
+			return false
+		}
+	}
+
+	return true
+}
+
+func preUpdate(ctx context.Context, cr *svcapitypes.DBSubnetGroup, obj *svcsdk.ModifyDBSubnetGroupInput) error {
+	obj.DBSubnetGroupName = awsclient.String(meta.GetExternalName(cr))
+	return nil
+}
+
+func (e *hooks) postUpdate(ctx context.Context, cr *svcapitypes.DBSubnetGroup, resp *svcsdk.ModifyDBSubnetGroupOutput, upd managed.ExternalUpdate, err error) (managed.ExternalUpdate, error) {
+	if err != nil {
+		return upd, err
+	}
+
+	cr.Status.SetConditions(v1.Available())
+	return upd, svcutils.UpdateTagsForResource(e.client, cr.Spec.ForProvider.Tags, resp.DBSubnetGroup.DBSubnetGroupArn)
+}
+
+func preCreate(ctx context.Context, cr *svcapitypes.DBSubnetGroup, obj *svcsdk.CreateDBSubnetGroupInput) error {
+	obj.DBSubnetGroupName = awsclient.String(meta.GetExternalName(cr))
+	return nil
+}
+
+func postCreate(ctx context.Context, cr *svcapitypes.DBSubnetGroup, resp *svcsdk.CreateDBSubnetGroupOutput, cre managed.ExternalCreation, err error) (managed.ExternalCreation, error) {
+	if err != nil {
+		return cre, err
+	}
+
+	meta.SetExternalName(cr, awsclient.StringValue(resp.DBSubnetGroup.DBSubnetGroupName))
+	cre.ExternalNameAssigned = true
+	return cre, nil
+}
```

Comment:
```suggestion
```

You don't need to capture the external name because we fill it in external name initializer and are able to use that value. So, you can always assume it exists.

Reply:
Fixed